### PR TITLE
Make elastic timeout configurable for HorovovJob.

### PR DESCRIPTION
**Pull request by supercharleszhu**
## Why are the changes needed?
The default elastic timeout to wait for worker pod to be ready is 60s https://github.com/horovod/horovod/blob/55a73d4f662bf637f6994a61f62b517ad9554dc1/horovod/runner/launch.py#L452C32-L452C51

This PR is to make it configurable

## What changes were proposed in this pull request?
Changes:
* Increased elastic_timeout to 20 min;
* make elastic_timeout configurable by user

## Check all the applicable boxes 

- [ X ] I updated the documentation accordingly.
- [ X ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

**Review by ByronHsu — APPROVED** [review]

**Review by ByronHsu — DISMISSED** [review]
LGTM. cc @Future-Outlier @pingsutw

File: plugins/flytekit-kf-mpi/flytekitplugins/kfmpi/task.py [hunk 1]
```diff
@@ -233,6 +233,7 @@ class HorovodJob(object):
         verbose: Optional flag indicating whether to enable verbose logging (default: False).
         log_level: Optional string specifying the log level (default: "INFO").
         discovery_script_path: Path to the discovery script used for host discovery (default: "/etc/mpi/discover_hosts.sh").
+        elastic_timeout: horovod eslastic timeout in second (default: 1200).
```
Comment (eapolinario) [inline plugins/flytekit-kf-mpi/flytekitplugins/kfmpi/task.py:236]: Fix the typo (eslastic).

**Review by eapolinario — COMMENTED** [review]
Fix the typo and we're golden.

**Review by Future-Outlier — COMMENTED** [review]
@supercharleszhu 
Hi, before this is merge, can you provide screenshot with this branch's commit hash at remote execution?

this is my example: https://github.com/flyteorg/flytekit/pull/1870

**Comment by supercharleszhu** [comment]
updated. @eapolinario 

> @supercharleszhu Hi, before this is merge, can you provide screenshot with this branch's commit hash at remote execution?
> 
> this is my example: #1870

@Future-Outlier we are from LinkedIn and committed and tested in our internal branch. The commit hash is not sync with OSS. Let me know screenshot like this make sense or not? (masked some of the internal settings)

WX20240807-203835@2x

**Review by pingsutw — APPROVED** [review]

**Pull request merged**